### PR TITLE
Remove mentions of Rotor from codebase

### LINKED ISSUE #9872: [Question] Mentions of 'Rotor' in the source code
I don't know if it matters or not, but there's a few mentions of ['Rotor' in the source](https://github.com/dotnet/coreclr/search?q=Rotor&utf8=%E2%9C%93), mostly under `\src\pal\`, `\src\inc\` and `\src\vm\`. 

Should they be updated/removed or just left in there for historical reasons?

### PR DESCRIPTION
**Pull request by AustinWise**
Many references to Rotor in this codebase are confusing, as they tend to contrast Rotor's behavior with .NET Framework (neither of which matches CoreCLR) or reference compatibility problems with 15 year old compilers.

This series of commits should remove all mentions rotor from the codebase, except from the many non-compiling PAL tests. Many of them reflect a time when the PAL was built as a shared library, so once they are cleaned up they will be purged of the reference to `librotor_pal`.

Fixes #9872

### CODE REVIEW

**Comment by jkotas** [comment]
Build break:

`...\src\inc\safemath.h(275): error C3861: '_ASSERTE': identifier not found (compiling source file ...\src\ildasm\ceeload.cpp) [...\bin\obj\Windows_NT.x64.Checked\src\ildasm\exe\ildasm.vcxproj]`

**Comment by AustinWise** [comment]
Interestingly, I did not have any trouble with building this on Linux. So that might mean linux builds have a few more asserts than Windows builds. Perhaps this is something worth investigating in the future?

I reverted my changes in safemath.h for now.

File: src/inc/safemath.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #define _ASSERTE_SAFEMATH _ASSERTE
 #else
 // Otherwise (eg. we're being used from a tool like SOS) there isn't much
-// we can rely on that is both available everywhere and rotor-safe.  In 
+// we can rely on that is both available everywhere.  In
```
[the remark this answers is not in the document]
Comment (AustinWise) [reply src/inc/safemath.h]: Done

File: src/inc/securitywrapper.h [hunk 2]
```diff
@@ -12,7 +12,9 @@
 #ifndef _SECURITY_WRAPPER_H
 #define _SECURITY_WRAPPER_H
 
-// This file should not even be included on Rotor.
+#ifdef FEATURE_PAL
```
[the remark this answers is not in the document]
Comment (AustinWise) [reply src/inc/securitywrapper.h]: Done.

File: src/inc/securitywrapper.h [hunk 3]
```diff
@@ -12,7 +12,9 @@
...
+#error This file should not be included on non-Windows platforms.
+#endif
```
[the remark this answers is not in the document]
Comment (AustinWise) [reply src/inc/securitywrapper.h]: It's gone now.

**Comment by jkotas** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test  please

**Pull request merged**